src/resource.c

```diff
@@ -745,7 +750,7 @@ coap_delete_observers(coap_context_t *context, coap_session_t *session) {
 }
 
 static void
-coap_notify_observers(coap_context_t *context, coap_resource_t *r) {
+coap_notify_observers(coap_context_t *context, coap_resource_t *r, int deleting) {
```

Comment:
To avoid non-descriptive numeric values being passed when calling this function, I would like to see a `#define` or—better—`enum` for this flag, e.g., `coap_notify_observers(.... COAP_NOTIFY_DELETE)`. The enum can be static, of course.

Reply:
How about using bool data type (need to include header file “stdbool.h” if not C++)?

Reply:
I have been avoiding this so far (merely for consistency). I am not completely against but from `coap_notify_observers(.... true)`, you also would not see what this flag is really about. I leave it up to you.

Reply:
OK, I will come up with an enum then.

Reply:
Code pushed using enum for clarity.